env.js

```diff
@@ -48,33 +48,59 @@ module.exports = (function () {
     maybeReplaceWithContentsOfFile(env.httpsConfig, 'pfx');
   }
   if (env.httpsPort != null && env.httpsConfig == null) {
-    throw new Error('No https config provided, please set HTTPS_CONFIG with at least the certificate to use.');
+    throw new Error(
+      'No https config provided, please set HTTPS_CONFIG with at least the certificate to use.'
+    );
   }
 
   if (env.httpPort == null && env.httpsPort == null) {
-    throw new Error('Must specify either PORT or HTTPS_PORT in your environment.');
+    throw new Error(
+      'Must specify either PORT or HTTPS_PORT in your environment.'
+    );
   }
 
   env.userApi = {
-    service: config.fromEnvironment('TIDEPOOL_AUTH_CLIENT_ADDRESS', 'shoreline:9107'),
+    service: config.fromEnvironment(
+      'TIDEPOOL_AUTH_CLIENT_ADDRESS',
+      'shoreline:9107'
+    ),
 
     // Name of this server to pass to user-api when getting a server token
-    serverName: config.fromEnvironment("SERVER_NAME", "jellyfish:default"),
+    serverName: config.fromEnvironment('SERVER_NAME', 'jellyfish:default'),
 
     // The secret to use when getting a server token from user-api
-    serverSecret: config.fromEnvironment("TIDEPOOL_SERVER_SECRET")
+    serverSecret: config.fromEnvironment('TIDEPOOL_SERVER_SECRET'),
   };
 
   env.gatekeeper = {
-    service: config.fromEnvironment('TIDEPOOL_PERMISSION_CLIENT_ADDRESS', 'gatekeeper:9123')
+    service: config.fromEnvironment(
+      'TIDEPOOL_PERMISSION_CLIENT_ADDRESS',
+      'gatekeeper:9123'
+    ),
   };
 
   env.seagull = {
-    service: config.fromEnvironment('TIDEPOOL_SEAGULL_CLIENT_ADDRESS', 'seagull:9120')
+    service: config.fromEnvironment(
+      'TIDEPOOL_SEAGULL_CLIENT_ADDRESS',
+      'seagull:9120'
+    ),
   };
 
   env.mongo = {
-    connectionString: cs('data')
+    connectionString: cs('data'),
+  };
+
+  // users who are going to upload via platform during migration
+  var platformUsers;
+  try {
+    usersFile = fs.readFileSync(__dirname + '/platform_users.json');
+    platformUsers = JSON.parse(usersFile);
```

Comment:
Do we want some check to make sure `platformUsers` is non-null and an array? Otherwise, usage in other functions will cause an failure.

Also, given that an exception here will cause the `platformUsers` to `[]`, I'm concerned if I go to update the file and make a mistake in the JSON, then this will silently fail and cause all users to go back to Jellyfish uploads (even if they've previous been uploading as a Platform user). That might be bad. Perhaps a hard fail if the file cannot be found, read, or parsed, or `platformUsers` is null or not an array?

What do you think?

Also, any thoughts on how to make this work for testing on a QA environment?

Reply:
Also, we should map all of the user ids to lowercase and trim whitespace (just in case).